From 9fcaa8826c180273829cf03e5fe34e08daef721f Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Mon, 12 Jul 2010 11:57:55 +0200
Subject: [PATCH] aggety: don't wipe CLOCAL flag

gettey should not clear the flag. It seems that the flag is
automatically enabled for serial consoles tht don't use CD signal.

Addresses: http://bugzilla.redhat.com/show_bug.cgi?id=598631
Signed-off-by: Karel Zak <kzak@redhat.com>
---
 login-utils/agetty.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/login-utils/agetty.c b/login-utils/agetty.c
index 39a1fd3..0cb00db 100644
--- a/login-utils/agetty.c
+++ b/login-utils/agetty.c
@@ -706,7 +706,7 @@ termio_init(tp, speed, op)
     /* flush input and output queues, important for modems! */
     (void) tcflush(0, TCIOFLUSH);
 
-    tp->c_cflag = CS8 | HUPCL | CREAD;
+    tp->c_cflag = CS8 | HUPCL | CREAD | (tp->c_cflag & CLOCAL);
     cfsetispeed(tp, speed);
     cfsetospeed(tp, speed);
     if (op->flags & F_LOCAL) {
-- 
1.6.6.1

